zephyr/lib/pm_runtime.c

```diff
@@ -89,19 +90,18 @@ void platform_pm_runtime_disable(uint32_t context, uint32_t index)
 	}
 }
 
-void platform_pm_runtime_init(struct pm_runtime_data *prd)
-{ }
-
-void platform_pm_runtime_get(enum pm_runtime_context context, uint32_t index,
-			     uint32_t flags)
-{ }
-
-void platform_pm_runtime_put(enum pm_runtime_context context, uint32_t index,
-			     uint32_t flags)
-{ }
-
-void platform_pm_runtime_prepare_d0ix_en(uint32_t index)
-{ }
-
-void platform_pm_runtime_power_off(void)
-{ }
+/** Is the _power_ active. The power, not its management. */
+bool pm_runtime_is_active(enum pm_runtime_context context, uint32_t index)
```

Comment:
Might be completely wrong here (never messed around with the PM-related stuff) but won't this return a different result compared with our implementation of `platform_pm_runtime_is_active` (e.g: from `imx8`)? Our implementation looks like:

```
-static inline bool platform_pm_runtime_is_active(uint32_t context,
-                                                uint32_t index)
-{
-       return false;
-}
```

meaning the function will always return `false`, while the new function will return `true` if called with `PM_RUNTIME_DSP` context.

Reply:
@LaurentiuM1234 This is a good contender for the most confusing interface in SOF :D, but OTOH, the semantics match the Linux runtime-pm  pm_runtime_active(), so it's at least in good company. The documentation for both states that this should return the state of the targetted resource, i.e. "return true if the resource is active or pm disabled, false otherwise.". So I'd say your dummy implementation is actually not correct. If PM is disabled (as is the case if the interface is not implemented, there is no runtime control done of power via this interface), this should return "true" indicating that the resource is powered on always.